[Python] test_address_resolver_with_fake_address

## Description of the failing test

**Test Name:** test_address_resolver_with_fake_address
**Test Location:** tests/sync_tests/test_sync_address_resolver.py
**Failure Permlink:** https://github.com/valkey-io/valkey-glide/actions/runs/26383200704/job/77656333432#logs
**Frequency:** Observed in scheduled CI run on 2026-05-25. Failed in ALL 72 Python jobs (systemic).

## Logs

```
FAILED tests/sync_tests/test_sync_address_resolver.py::TestSyncAddressResolver::test_address_resolver_with_fake_address[True]
- glide_shared.exceptions.ClosingError: Failed to create initial connections - IoError: Failed to refresh both connections - IoError: Node: "fake-host-that-does-not-exist.invalid:9999" received errors: `failed to lookup address information: Name or service not known`

FAILED tests/sync_tests/test_sync_address_resolver.py::TestSyncAddressResolver::test_address_resolver_with_fake_address[False]
- glide_shared.exceptions.ClosingError: Received error for address `fake-host-that-does-not-exist.invalid:9999`: timed out
```

## Additional Context

- **System Information:** GitHub Actions (ubuntu x86_64 and aarch64, amazon-linux containers)
- **Engine Versions:** All (6.2, 7.0, 7.2, 8.0, 8.1, 9.0)
- **Python Versions:** 3.9, 3.10, 3.11, 3.12, 3.13, 3.14
- **Note:** This test fails in ALL jobs, suggesting a systemic issue rather than a flaky test. The test expects a specific error when connecting to a fake address but gets a different error (timeout vs DNS lookup failure depending on parametrization).

*This issue was automatically created by the failing-tests-finder agent.*

Started failing in FMT since May 24:
May 24 failing Python FMT: [FMT Python #21](https://github.com/valkey-io/valkey-glide/actions/runs/26351816064)
May 23 passing Python FMT: [FMT Python #20](https://github.com/valkey-io/valkey-glide/actions/runs/26323191755)

--- a/python/glide-sync/glide_sync/_glide_ffi.py
+++ b/python/glide-sync/glide_sync/_glide_ffi.py
@@ -188,6 +188,7 @@ class _GlideFFI:
             );
 
             typedef uint16_t (*AddressResolverCallback)(
+                uintptr_t client_id,
                 const uint8_t* host,
                 size_t host_len,
                 uint16_t port,
@@ -217,7 +218,8 @@ class _GlideFFI:
                 size_t connection_request_len,
                 const ClientType* client_type,
                 PubSubCallback pubsub_callback,
-                AddressResolverCallback address_resolver
+                AddressResolverCallback address_resolver,
+                uintptr_t client_id
             );
             void close_client(const void* client_adapter_ptr);
             void free_connection_response(ConnectionResponse* connection_response_ptr);

--- a/python/glide-sync/glide_sync/glide_client.py
+++ b/python/glide-sync/glide_sync/glide_client.py
@@ -153,7 +153,7 @@ class BaseClient(CoreCommands):
             client_type,
             pubsub_callback,
             address_resolver_callback,
-            0,
+            0,  # client_id is not used by the Python client
         )
 
         Logger.log(Level.INFO, "connection info", "new connection established")

--- a/python/glide-sync/glide_sync/glide_client.py
+++ b/python/glide-sync/glide_sync/glide_client.py
@@ -121,6 +121,7 @@ class BaseClient(CoreCommands):
             resolver_fn = self._config.address_resolver
 
             def _address_resolver_callback(
+                client_id,
                 host_ptr,
                 host_len,
                 port,
@@ -152,6 +153,7 @@ class BaseClient(CoreCommands):
             client_type,
             pubsub_callback,
             address_resolver_callback,
+            0,
         )
 
         Logger.log(Level.INFO, "connection info", "new connection established")

